packages/react-native/Libraries/Text/Text/RCTTextView.mm

```diff
@@ -62,6 +62,7 @@ @implementation RCTTextView {
 
   id<RCTEventDispatcherProtocol> _eventDispatcher; // [macOS]
   NSArray<RCTUIView *> *_Nullable _descendantViews; // [macOS]
+  RCTUIView *_Nullable _currentHoveredSubview; // [macOS]
```

Comment:
expressed thoughts offline about tracking the path of hovered subviews; that said I wouldn't over-index on optimizing to not send Native->JS events for subviews that do not have listeners registered on native, they will no-op on the JS thread

Reply:
On a related note, I did some more testing with different layers having/not having mouse events and found some additional bugs. On the bright side, I've been able to get shadow view traversal to work, and that fixes all the bugs that I found, so that's definitely a big plus!